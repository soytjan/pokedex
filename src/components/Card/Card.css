.Card {
  max-height: 300px;
  overflow: scroll;
  border: 1px solid darkgray;
  box-shadow: 5px 5px 4px gray;
  max-width: 490px;
}

.Card:hover {
  cursor: pointer;
  background-color: lightblue;
}

.card-header {
  max-height: 78px;
}

.capitalize {
  text-transform: capitalize;
}