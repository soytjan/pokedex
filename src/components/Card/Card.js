import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

// need to pass down a handleClick function from CardDeck
// need to check and see isOpened = true || false


const Card = ({pokemon}) => {
  return (
    <article className='Card'>
      <h3>{pokemon.name}</h3> 
    </article>
  )
}

Card.propTypes = {
  pokemon: PropTypes.object,
};

export default Card;
