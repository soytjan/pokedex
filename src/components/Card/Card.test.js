import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  let renderedComponent;
  let mockPokemon;

  beforeAll(() => {
    mockPokemon = {
      "id": "1",
      "name": "normal",
      "pokemon": [
        "16",
        "17",
        "18",
        "19",
        "20"
      ]
    };
  })

  beforeEach(() => {
    renderedComponent = shallow(
      <Card 
        pokemon={mockPokemon}
        key={mockPokemon.id}
      />
    )
  })

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  })
})