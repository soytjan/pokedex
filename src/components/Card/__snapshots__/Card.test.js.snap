// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    onClick={[MockFunction]}
    pokemon={
      Object {
        "id": "1",
        "name": "normal",
        "pokemon": Array [
          "16",
          "17",
          "18",
          "19",
          "20",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="capitalize card-header"
        >
          normal
        </h2>,
        null,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "normal",
          "className": "capitalize card-header",
        },
        "ref": null,
        "rendered": "normal",
        "type": "h2",
      },
      null,
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="capitalize card-header"
          >
            normal
          </h2>,
          null,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "normal",
            "className": "capitalize card-header",
          },
          "ref": null,
          "rendered": "normal",
          "type": "h2",
        },
        null,
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot when toggled 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "id": "1",
              "name": "normal",
              "pokemon": Array [
                "16",
                "17",
                "18",
                "19",
                "20",
              ],
            },
          ],
        ],
      }
    }
    pokemon={
      Object {
        "id": "1",
        "name": "normal",
        "pokemon": Array [
          "16",
          "17",
          "18",
          "19",
          "20",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="capitalize card-header"
        >
          normal
        </h2>,
        null,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "normal",
          "className": "capitalize card-header",
        },
        "ref": null,
        "rendered": "normal",
        "type": "h2",
      },
      null,
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="capitalize card-header"
          >
            normal
          </h2>,
          null,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "normal",
            "className": "capitalize card-header",
          },
          "ref": null,
          "rendered": "normal",
          "type": "h2",
        },
        null,
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
